### PR TITLE
feat: added path string navigation (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by jspears**
Was looking at new typescript features and thought this might
be a fun little addition.   I haven't used this library much so, but
am very interested.

This PR allows for `Optic.path` to use a typesafe path with dot 
delineation.   Very similar to lodash.get.   If interested I can put a 
bit effort into it, if not well no problem.

Any feedback is welcome.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,8 +34,8 @@
     "doctoc": "^1.4.0",
     "jest": "^26.0.1",
     "prettier": "^2.0.2",
-    "ts-jest": "^26.0.0",
+    "ts-jest": "^26.4.1",
     "ts-node": "^8.8.1",
-    "typescript": "^3.8.3"
+    "typescript": "beta"
   }
```
Comment (jspears) [inline package.json:40]: Need version 4.1 for this to work

File: README.md [hunk 2]
```diff
@@ -148,6 +148,12 @@ const bar = foo.prop('bar')
 const bar = O.optic<Data>().path(['foo', 'bar'])
 ```
 
+Or
+```typescript
+const bar = O.optic<Data>().prop('foo.bar')
+```
+
```
Comment (jspears) [inline README.md:155]: of course its just prop -- we could think more about various ones later.

File: src/internals.ts [hunk 3]
```diff
@@ -465,7 +466,12 @@ export class Optic {
...
-    return new Optic(compose(this._ref, prop(key)))
+    const parts = key.split('.')
+    if (parts.length === 1) {
+      return new Optic(compose(this._ref, prop(key)))
+    } else {
+      return this.path(parts)
+    }
   }
 
   path(keys: string[]): Optic {
```
Comment (jspears) [inline src/internals.ts:477]: This could be made typesafe

File: tests/optic.spec.ts [hunk 4]
```diff
@@ -140,6 +140,19 @@ describe('lens/path', () => {
...
+  const source: Source = { foo: { bar: { baz: 'quux' } }, xyzzy: 42 }
+
+  const lens = O.optic_<Source>().prop('foo.bar.baz')
+  type Focus = string
+
+  it('get', () => {
+    const result: Focus = O.get(lens)(source)
+    expect(result).toEqual('quux')
+  })
+
```
Comment (jspears) [inline tests/optic.spec.ts:154]: obviously not throughly tested.

File: tests/optic.spec.ts [hunk 5]
```diff
@@ -140,6 +140,19 @@ describe('lens/path', () => {
     })
   })
 })
+describe('lens/path/prop', () => {
+  type Source = { foo: { bar: { baz: string } }; xyzzy: number }
+  const source: Source = { foo: { bar: { baz: 'quux' } }, xyzzy: 42 }
+
+  const lens = O.optic_<Source>().prop('foo.bar.baz')
+  type Focus = string
```
Comment (jspears) [inline tests/optic.spec.ts:148]: This is the magic

**Comment by akheron** [comment]
Nice! This is a good idea. 

Instead of changing `prop()`, it would make more sense to add a new overload for `path()` so that it would work like this when given a single string. 

Right now I can’t think of any other optics this new string magic would be useful for, but I’m surely interested if you have more ideas!

**Comment by jspears** [comment]
Closing in preference to https://github.com/akheron/optics-ts/pull/12

**Pull request closed without merging**